handler/common.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE
+
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+const (
+	mimeTypeApplicationJSON = "application/json"
+)
+
+// NotSupportdResponse is data for a method not supported response
+type NotSupportdResponse struct {
+	Error            string
+	MethodName       string
+	URI              string
+	SupportedMethods []string
+}
+
+// MethodNotSupportedResponse writes a method not supported response to the writer
```

Comment:
nit: perhaps add periods to the end of these comments. It's the comment convention recommended in the Go wiki and we've been trying to migrate code to use it, though we're definitely far from that mark :) Can start clean here at least